classlib/src/main/java/org/teavm/classlib/java/util/THashMap.java

```diff
@@ -237,6 +238,23 @@ private static boolean valuesEq(TMap.Entry<?, ?> entry, TMap.Entry<?, ?> oEntry)
         public TIterator<TMap.Entry<K, V>> iterator() {
             return new EntryIterator<>(associatedMap);
         }
+
+        @Override
+        public void forEach(Consumer<? super Entry<K, V>> action) {
+            if (associatedMap.elementCount > 0) {
+                int prevModCount = associatedMap.modCount;
+                for (int i = 0; i < associatedMap.elementData.length; i++) {
+                    HashEntry<K, V> entry = associatedMap.elementData[i];
+                    while (entry != null) {
+                        action.accept(entry);
+                        entry = entry.next;
+                    }
+                }
+                if (prevModCount != associatedMap.modCount) {
```

Comment:
I think this check should be done after every `accept` call. At least, this behaviour will be consistent with default `forEach` implementation from `Iterable` class.

Reply:
Did it.